robottelo/hosts.py

```diff
@@ -359,3 +366,121 @@ def patch_os_release_version(self, distro=DISTRO_RHEL7):
         else:
             raise ContentHostError('No distro package available to retrieve release version')
         return self.execute(f"echo '{rh_product_os_releasever}' > /etc/yum/vars/releasever")
+
+
+class Capsule(Host):
+    def restart_services(self):
+        """Restart services, returning True if passed and stdout if not"""
+        result = self.execute('foreman-maintain service restart').status
+        return True if result.status == 0 else result.stdout
+
+    def check_services(self):
+        error_msg = 'Some services are not running'
+        result = self.execute('foreman-maintain service status')
+        if result.status == 0:
+            return True
+        for line in result.stdout.splitlines():
+            if error_msg in line:
+                return line.replace(error_msg, '').strip()
+
+    def installer(self, **extra_kwargs):
+        """General purpose installer"""
+        command = f'satellite-installer --scenario {self.__class__.__name__.lower()}'
+        extras = ' '.join(
+            [f'--{key.replace("_", "-")} {value}' for key, value in extra_kwargs.items()]
+        )
+        return self.execute(f'{command} {extras}')
+
+
+class Satellite(Capsule):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._init_nailgun()
+        self._init_cli()
+        self._init_airgun()
+
+    def _init_nailgun(self):
```

Comment:
These could be cached_property decorated properties, supplying `self.api`, `self.cli`, and `self.ui_session`

Reply:
I don't believe that will work with the way we populate api and cli.